plugins/storage/volume/solidfire/src/main/java/org/apache/cloudstack/storage/datastore/util/SolidFireUtil.java

```diff
@@ -683,19 +694,58 @@ private static SolidFireVag getVagForRandomHostInCluster(List<SolidFireUtil.Soli
         return null;
     }
 
-    public static void placeVolumeInVolumeAccessGroups(SolidFireConnection sfConnection, long sfVolumeId, List<HostVO> hosts) {
+    public static void placeVolumeInVolumeAccessGroups(SolidFireConnection sfConnection, long sfVolumeId, List<HostVO> hosts, String clusterUuId) {
```

Comment:
Do we want to pass in a List<HostVO> still if we can now retrieve that info from the cluster DAO using the cluster UUID?

Reply:
we need the list of hosts to build the vag to iqn map, we need the hostDao to get the list of hosts. I think it is better to just pass the list of hosts in this case.

Reply:
Either way...my initial thoughts were that you could derive the list of hosts from using the cluster ID with the cluster DAO.